src/hooks/useActiveDescendantListFocus/__tests__/useActiveDescendantListFocus.jest.js

```diff
@@ -5,32 +5,109 @@ import userEvent from "@testing-library/user-event";
-const ITEM_IDS = [FIRST_ITEM_ID, SECOND_ITEM_ID];
-
-function renderHookForTest({ onItemClick = jest.fn(), isItemSelectable = () => true, isHorizontal = false }) {
+const THIRD_ITEM_ID = "id-3";
+const FOURTH_ITEM_ID = "id-4";
+const FIFTH_ITEM_ID = "id-5";
+const ITEM_IDS = [FIRST_ITEM_ID, SECOND_ITEM_ID, THIRD_ITEM_ID, FOURTH_ITEM_ID, FIFTH_ITEM_ID];
+
+function renderHookForTest({
+  onItemClick = jest.fn(),
+  keepOptionSelected = false,
+  isItemSelectable = () => true,
+  isHorizontal = false
+}) {
   element = document.createElement("div");
   element.tabIndex = -1; // some tests focus the element - a tabIndex value is required for updating the document.activeIndex value
   document.body.appendChild(element);
 
-  return renderHook(() =>
-    useActiveDescendantListFocus({
-      focusedElementRef: {
-        current: element
-      },
-      itemsIds: ITEM_IDS,
-      isItemSelectable: isItemSelectable,
-      onItemClick,
-      isHorizontalList: isHorizontal
-    })
-  );
+  const props = {
+    focusedElementRef: {
+      current: element
+    },
+    keepOptionSelected,
+    itemsIds: ITEM_IDS,
+    isItemSelectable: isItemSelectable,
+    onItemClick,
+    isHorizontalList: isHorizontal
+  };
+  return renderHook(argprops => useActiveDescendantListFocus({ ...props, ...argprops }));
 }
 
-function runListUnitTest(isHorizontal) {
+function runListUnitTest({ isHorizontal, keepOptionSelected }) {
   const moveForwardKey = isHorizontal ? "{arrowRight}" : "{arrowDown}";
   const oppositeMoveForwardKey = !isHorizontal ? "{arrowRight}" : "{arrowDown}";
+  const naturalIndex = keepOptionSelected ? 0 : undefined;
+  const naturalIndexAfterInitial = keepOptionSelected ? 0 : -1;
+
+  it("should focus same item after item changed", async () => {
+    const onItemClick = jest.fn();
+    const { result, rerender } = renderHookForTest({ onItemClick, isHorizontal, keepOptionSelected });
+
+    act(() => {
+      // set focus on the list's element which in charge on natural focus element
+      element.focus();
+    });
+    act(() => {
+      userEvent.keyboard(moveForwardKey);
+    });
+    act(() => {
+      userEvent.keyboard(moveForwardKey);
+    });
+    act(() => {
+      userEvent.keyboard(moveForwardKey);
+    });
+
+    const beforeIndex = result.current.visualFocusItemIndex;
+    const beforeId = result.current.visualFocusItemId;
+
+    const removeItems = 2;
+    const itemsAfterRemove = ITEM_IDS.slice(removeItems);
+    rerender({ itemsIds: itemsAfterRemove });
+
+    if (!keepOptionSelected) {
+      expect(result.current.visualFocusItemId).toEqual(itemsAfterRemove[beforeIndex]);
+      expect(result.current.visualFocusItemIndex).toEqual(beforeIndex);
+    } else {
+      expect(result.current.visualFocusItemId).toEqual(beforeId);
+      expect(result.current.visualFocusItemIndex).toEqual(beforeIndex - removeItems);
+    }
+  });
+
+  it("should focus correct item on focus", async () => {
+    const onItemClick = jest.fn();
```

Comment:
if u expect here to 2 different results for 2 different state it should be 2 different states and the name of each test should indicate what "correct item is".
For example (new names for the 2 different tests):
- "Should not visually focus on any item after initial natural focus on the list wrapper element."
- "Should visually focus on the first item in the list after initial natural focus on the list wrapper element."